src/main/java/com/limechain/runtime/SharedMemory.java

```diff
@@ -55,7 +55,7 @@ public RuntimePointerSize writeData(byte[] data) {
     public void writeData(byte[] data, RuntimePointerSize runtimePointerSize) {
         ByteBuffer memoryBuffer = memory.buffer();
         memoryBuffer.position(runtimePointerSize.pointer());
-        memoryBuffer.put(data, 0, runtimePointerSize.size());
+        memoryBuffer.put(data, 0, Math.min(data.length, runtimePointerSize.size()));
```

Comment:
For some reason the Runtime was panicking when trying to write the data to a given pointer-size. In the given usecase the data's length was smaller than the size of the pointer. After this change the runtime stopped panicking, but when I think about it now it might cause issues in some use cases